f3/api.go

```diff
@@ -34,7 +34,7 @@ type Receiver interface {
 type Network interface {
 	// Sends a message to all other participants.
 	// The message's sender must be one that the network interface can sign on behalf of.
-	Broadcast(msg *GMessage)
+	Broadcast(msg *GMessage) error
```

Comment:
This is a client failures ,and F3 can't do anything with it. It will just end up back at the client as a result from one of the few entry points where the client calls F3. Yes the F3 instance internal state is still consistent, but the protocol execution may have a problem.

Whatever the client would do in response to this error from F3, it should have done it before sending the error to F3 in the first place. E.g. retrying any transient network failure, etc. If it can't do anything, then it still can't do anything when this is thrown back at it. If it would, say, tear down the instance and wait for a finality certificate, just do that right away. If it would shrug and say "well lets just continue and hope the protocol is fine", then just continue and don't return an error in the first place.

I could well be missing something here, since nothing returns an error into this interface yet, but where do you have in mind that this error can be useful?

(Can you separate this change from the rest of the PR, which is otherwise an improvement)?

Reply:
I can for sure separate.
The only way to tear down the instance would be to panic within Broadcast.
One of the reasons I've added it was the possibility of network failures and letting the instance know about them, but I think you are right; it doesn't need to know.